### PR TITLE
Remove unused variables in Java files

### PR DESCRIPTION
**Pull request by chan-j-d**
This removes or utilizes unused fields and local variables according to Intellij's analysis. 
A follow-up PR will be for removing unused methods. 

Commit message:
```
There are some unused fields and variables in the Java source code,
which adds some unnecessary complexity and makes the code harder to
read.

Let's clean up these unused fields and local variables.
```

### CODE REVIEW

**Review by yhtMinceraft1010X — CHANGES_REQUESTED** [review]

File: src/test/java/reposense/parser/ArgsParserTest.java [hunk 1]
```diff
@@ -57,7 +57,6 @@
     private static final String TEST_REPO_DELTA = "https://github.com/reposense/testrepo-Delta.git";
 
     private static final String DEFAULT_TIMEZONE = "Asia/Singapore";
-    private static final ZoneId TIME_ZONE_ID = TestUtil.getZoneId(DEFAULT_TIMEZONE);
```
[the remark this answers is not in the document]
Comment (chan-j-d) [reply src/test/java/reposense/parser/ArgsParserTest.java:60]: Thanks for catching this!

File: src/main/java/reposense/parser/ArgsParser.java [hunk 2]
```diff
@@ -67,25 +67,12 @@
...
-            "\"Since Date\" cannot be later than \"Until Date\".";
-    private static final String MESSAGE_SINCE_DATE_LATER_THAN_TODAY_DATE =
-            "\"Since Date\" must not be later than today's date.";
     private static final String MESSAGE_HAVE_SINCE_DATE_UNTIL_DATE_AND_PERIOD =
             "\"Since Date\", \"Until Date\", and \"Period\" cannot be applied together.";
     private static final String MESSAGE_USING_DEFAULT_CONFIG_PATH =
             "Config path not provided, using the config folder as default.";
     private static final String MESSAGE_INVALID_CONFIG_PATH = "%s is malformed.";
     private static final String MESSAGE_INVALID_CONFIG_JSON = "%s Ignoring the report config provided.";
-    private static final String MESSAGE_IGNORING_REPORT_CONFIG_SINCE_DATE =
```
[the remark this answers is not in the document]
Comment (chan-j-d) [reply src/main/java/reposense/parser/ArgsParser.java:80]: Yeah I noticed it. But it seems somewhat strange that this made it in to the code base even though the actual feature is likely a long way off. I think we should remove it here and add only add it in when #634 is completed.

File: src/main/java/reposense/parser/ArgsParser.java [hunk 3]
```diff
@@ -67,25 +67,12 @@
...
-            "Ignoring \"Since Date\" in report config, it has already been provided as a command line argument.";
-    private static final String MESSAGE_IGNORING_REPORT_CONFIG_UNTIL_DATE =
-            "Ignoring \"Until Date\" in report config, it has already been provided as a command line argument.";
-    private static final String MESSAGE_IGNORING_REPORT_CONFIG_PERIOD =
-            "Ignoring \"Period\" in report config, it has already been provided as a command line argument.";
-    private static final String MESSAGE_IGNORING_REPORT_SINCE_DATE_UNTIL_DATE_AND_PERIOD =
-            "Ignoring \"Since Date\", \"Until Date\" and/or \"Period\" in report config as they cannot be applied "
-                    + "together.";
```
[the remark this answers is not in the document]
Comment (chan-j-d) [reply src/main/java/reposense/parser/ArgsParser.java:88]: > While reviewing this, I noticed that line 270 in the same file suggests that the --repos flag will override the --config flag, although the User Guide cli.md only states that both flags cannot be used with each other.

Interesting. It's something I just assumed was there. I think it'll be more appropriate in a separate PR.

[the remark this answers is not in the document]
Comment (yhtMinceraft1010X) [reply src/main/java/reposense/parser/ArgsParser.java:88]: In that case, I'll raise it up as a separate issue.

**Comment by chan-j-d** [comment]
@yhtMinceraft1010X I've updated accordingly.

**Review by yhtMinceraft1010X — CHANGES_REQUESTED** [review]
Nice job with the refactoring.

For spelling consistency, I recommend choosing either `TIME_ZONE` or `TIMEZONE` and also renaming `DEFAULT_TIMEZONE` as `DEFAULT_TIMEZONE_STRING`, since someone might think that `DEFAULT_TIMEZONE` is the `ZoneId` object rather than the `String`.

**Comment by chan-j-d** [comment]
> For spelling consistency, I recommend choosing either `TIME_ZONE` or `TIMEZONE` and also renaming `DEFAULT_TIMEZONE` as `DEFAULT_TIMEZONE_STRING`, since someone might think that `DEFAULT_TIMEZONE` is the `ZoneId` object rather than the `String`.

Makes sense, I've updated accordingly

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Comment by chan-j-d** [comment]
oops, active-developers, please ignore this

**Review by dcshzj — APPROVED** [review]
LGTM! This should not affect the logic of the software, so I'm going to directly merge this.

**Pull request merged**